test/sql/aggregate_groupby_sql_test.cpp

```diff
@@ -0,0 +1,164 @@
+                                         {"5|3|5|5|5",
+                                          "2|2|4|4|4"});
+
+
+  // free the database just created
+  txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+}
+
+TEST_F (AggregateGroupBySQLTests, AggregateGroupByMixedAVGsSQLTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+
+  CreateAndLoadTable();
+
+  // This test especially tests several AVGs in a row
+  ExecuteSQLQueryAndCheckUnorderedResult("SELECT SUM(a), AVG(a), COUNT(b), AVG(b), MAX(c), AVG(c) FROM test GROUP BY d;",
+                                         {"15|5|3|3|6|5",
+                                          "6|2|3|2|6|4"});
+
+
+  // free the database just created
+  txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+}
+
+TEST_F (AggregateGroupBySQLTests, AggregateGroupByAllAggregationsSQLTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+
+  CreateAndLoadTable();
+
+  ExecuteSQLQueryAndCheckUnorderedResult("SELECT AVG(a), SUM(a), MAX(a), MIN(a), COUNT(a) FROM test GROUP BY d;",
+                                         {"2|6|3|1|3",
+                                          "5|15|6|4|3"});
+
+
+  // free the database just created
+  txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+}
+
+TEST_F (AggregateGroupBySQLTests, AggregateGroupBySingleRowPerGroupSQLTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+
+  CreateAndLoadTable();
+
+  // This test especially tests several AVGs in a row
+  ExecuteSQLQueryAndCheckUnorderedResult("SELECT COUNT(*), MIN(b), MAX(c), AVG(d) FROM test GROUP BY a;",
+                                         {"1|4|6|2",
+                                          "1|2|3|1",
+                                          "1|3|6|2",
+                                          "1|2|3|2",
+                                          "1|2|3|1",
+                                          "1|2|6|1"});
+
+
+  // free the database just created
+  txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
```

Comment:
The tests are good. Does it make sense to move them into the aggregate_sql_test?